cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/CassandraDialect.java

```diff
@@ -210,12 +236,61 @@ public void removeTuple(EntityKey key, TupleContext tupleContext) {
 
 	@Override
 	public Association getAssociation(AssociationKey key, AssociationContext associationContext) {
-		return null; // TODO
+
+		String table = key.getTable();
+		Table tableMetadata = provider.getMetaDataCache().get( key.getTable() );
+		List<Column> tablePKCols = (List<Column>) tableMetadata.getPrimaryKey().getColumns();
+
+		Select select = select().all().from( quote( key.getTable() ) );
+		Select.Where selectWhere = select.where( eq( quote( key.getColumnNames()[0] ), QueryBuilder.bindMarker() ) );
+		for ( int i = 1; i < key.getColumnNames().length; i++ ) {
+			selectWhere = selectWhere.and( eq( quote( key.getColumnNames()[i] ), QueryBuilder.bindMarker() ) );
+		}
+
+		boolean requiredFiltering = false;
+		for ( Column column : tablePKCols ) {
+			String name = column.getName();
+			boolean foundColumn = false;
+			for ( int i = 0; i < key.getColumnNames().length; i++ ) {
+				if ( name.equals( key.getColumnNames()[i] ) ) {
+					foundColumn = true;
+					break;
+				}
+			}
+			if ( !foundColumn ) {
+				requiredFiltering = true;
+				break;
+			}
+		}
+
+		if ( requiredFiltering ) {
+			select.allowFiltering();
+		}
+
+		Object[] columnValues = key.getColumnValues();
+		ResultSet resultSet = bindAndExecute( columnValues, select );
+
+		if ( resultSet.isExhausted() ) {
+			return null;
+		}
+
+		Association association = new Association(
+				new ResultSetAssociationSnapshot(
+						key,
+						resultSet,
+						provider.getMetaDataCache()
```

Comment:
In `ResultSetAssociationSnapshot`, why is it that you add the columns from `tableMetadata.getPrimaryKey()`? I'd expect `key.getColumnNames()` to contain everything?